pkg/gcmanager/tracePod_worker.go

```diff
@@ -53,6 +53,28 @@ func (s *SpiderGC) handlePodEntryForTracingTimeOut(podEntry *PodEntry) {
 		if time.Now().UTC().After(podEntry.TracingStopTime) {
 			logger.With(zap.Any("podEntry tracing-reason", podEntry.PodTracingReason)).
 				Sugar().Infof("pod '%s/%s' is out of time, begin to gc IP", podEntry.Namespace, podEntry.PodName)
+
+			// If the statefulset application quickly experiences scaling down and up,
+			// check whether `Status.PodIPs` is empty to determine whether the Pod in the current K8S has completed the normal IP release to avoid releasing the wrong IP.
+			ctx := context.TODO()
+			currentPodYaml, err := s.podMgr.GetPodByName(ctx, podEntry.Namespace, podEntry.PodName, constant.UseCache)
+			if nil != err {
+				tracingReason := fmt.Sprintf("pod '%s/%s' is out of time, get the current pod status in K8S", podEntry.Namespace, podEntry.PodName)
+				if apierrors.IsNotFound(err) {
+					logger.With(zap.Any("podEntry tracing-reason", tracingReason)).
+						Sugar().Debugf("pod '%s/%s' not found", podEntry.Namespace, podEntry.PodName)
+					return
+				}
+
+				logger.With(zap.Any("podEntry tracing-reason", tracingReason)).
+					Sugar().Errorf("failed to get pod '%s/%s', error: %v", podEntry.Namespace, podEntry.PodName, err)
+				return
```

Comment:
// comment: the pod will be handled next time ?
return

Reply:
logger.With(zap.Any("podEntry tracing-reason", tracingReason)).
						Sugar().Errorf("failed to get pod '%s/%s', error: %v", podEntry.Namespace, podEntry.PodName, err)
					// the pod will be handled next time.
					return